src/cmake-tools.ts

```diff
@@ -920,9 +912,10 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
   /**
    * Implementation of `cmake.setVariant`
    */
-  async setVariant() {
-    const ret = await this._variantManager.selectVariant();
-    if (ret) {
+  async setVariant(name?: string) {
+    // ! for compatibility with return code style...
```

Comment:
I changed it from success (true) / failure (false) pattern to return code (0 for success) pattern. So the previous naming was inappropriate, and the current name should be fine. Can you see a better name?

Reply:
Yeah I did notice this. I did it this way because there was some timing issue with
```
    const ret = await this._variantManager.selectVariant(name);
    if (ret) {
      await this.configure();
    }
    return !ret;
```
Seems like that return line can bypass the awaits... I can try your ways though.